homeassistant/components/smartthings/cover.py

```diff
@@ -32,114 +30,101 @@
+            client,
+            device,
+            {
+                capability,
+                Capability.BATTERY,
+                Capability.WINDOW_SHADE_LEVEL,
+                Capability.SWITCH_LEVEL,
+            },
+        )
+        self.capability = capability
         self._attr_supported_features = (
             CoverEntityFeature.OPEN | CoverEntityFeature.CLOSE
         )
-        if (
-            Capability.switch_level in device.capabilities
-            or Capability.window_shade_level in device.capabilities
-        ):
+        if self.supports_capability(
+            Capability.SWITCH_LEVEL
+        ) or self.supports_capability(Capability.WINDOW_SHADE_LEVEL):
             self._attr_supported_features |= CoverEntityFeature.SET_POSITION
 
-        if Capability.door_control in device.capabilities:
+        if self.supports_capability(Capability.DOOR_CONTROL):
             self._attr_device_class = CoverDeviceClass.DOOR
-        elif Capability.window_shade in device.capabilities:
+        elif self.supports_capability(Capability.WINDOW_SHADE):
             self._attr_device_class = CoverDeviceClass.SHADE
-        elif Capability.garage_door_control in device.capabilities:
-            self._attr_device_class = CoverDeviceClass.GARAGE
 
     async def async_close_cover(self, **kwargs: Any) -> None:
         """Close cover."""
-        # Same command for all 3 supported capabilities
-        await self._device.close(set_status=True)
-        # State is set optimistically in the commands above, therefore update
-        # the entity state ahead of receiving the confirming push updates
-        self.async_schedule_update_ha_state(True)
+        await self.execute_device_command(self.capability, Command.CLOSE)
 
     async def async_open_cover(self, **kwargs: Any) -> None:
         """Open the cover."""
-        # Same for all capability types
-        await self._device.open(set_status=True)
-        # State is set optimistically in the commands above, therefore update
-        # the entity state ahead of receiving the confirming push updates
-        self.async_schedule_update_ha_state(True)
+        await self.execute_device_command(self.capability, Command.OPEN)
 
     async def async_set_cover_position(self, **kwargs: Any) -> None:
         """Move the cover to a specific position."""
-        if not self.supported_features & CoverEntityFeature.SET_POSITION:
-            return
-        # Do not set_status=True as device will report progress.
-        if Capability.window_shade_level in self._device.capabilities:
-            await self._device.set_window_shade_level(
-                kwargs[ATTR_POSITION], set_status=False
-            )
-        else:
-            await self._device.set_level(kwargs[ATTR_POSITION], set_status=False)
+        await self.execute_device_command(
+            (
+                Capability.WINDOW_SHADE_LEVEL
+                if self.supports_capability(Capability.WINDOW_SHADE_LEVEL)
```

Comment:
Do we really need to check for support on each call?
Can we do it one time in the init or in _update_attr?